# 7.23 Flow Through Orifice Meters

A square edged orifice of 2" diameter is installed in a 4" schedule 40 pipe. A mercury manometer is connected between standard taps 1D upstream and 0.5D downstream.

a) Find the theoretical calibration constant for the meter when 60 deg F water flows through the pipe in the fully turbulent region. 

b) Find the flow rate of 60 deg F water when the mercury deflection is 4.4"

In [1]:
from fluids.units import *
P1 = 2*u.bar # assumed
T = 60*u.degF
NPS, D_pipe, Do_pipe, t = nearest_pipe(NPS=4, schedule=40)
mu = 1.1e-3*u.Pa*u.s # viscosity of water
rho = 62.364*u.lb/u.ft**3 # density of water
Do = 2*u.inch

In [2]:
# b)
dP = (4.4*u.inch_Hg - 4.4*u.inch_H2O_60F)
P2 = P1 - dP
m =  differential_pressure_meter_solver(D=D_pipe, rho=rho, mu=mu, k=1e20, D2=Do, P1=P1, P2=P2, 
                                       meter_type='ISO 5167 orifice', taps='D and D/2')
Q = (m/rho).to(u.gal/u.min)
Q

105.73900650508155 <Unit('gallon / minute')>

In [3]:
# a)
C, _ = differential_pressure_meter_C_epsilon(D=D_pipe, rho=rho, m=m, mu=mu, k=1e20, D2=Do, P1=P1, P2=P2,
                                       meter_type='ISO 5167 orifice', taps='D and D/2')
C, flow_coefficient(D_pipe, Do, C)

(0.6067029339232126 <Unit('dimensionless')>,
 0.6260667537160854 <Unit('dimensionless')>)

Crane TP 410 reports a flow coefficient of 0.625 and a flow rate of 106 gal/min, indicating the problem benefited from the iteration little.